api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java

```diff
@@ -110,80 +112,58 @@ public CohortReview saveCohortReview(CohortReview cohortReview) {
     @Transactional
     public void saveFullCohortReview(CohortReview cohortReview, List<ParticipantCohortStatus> participantCohortStatuses) {
         cohortReview = saveCohortReview(cohortReview);
-        participantCohortStatuses = saveParticipantCohortStatuses(participantCohortStatuses);
+        saveParticipantCohortStatuses(participantCohortStatuses);
     }
 
-    @Override
-    public List<ParticipantCohortStatus> saveParticipantCohortStatuses(List<ParticipantCohortStatus> participantCohortStatuses) {
+    /**
+     * This implementation manually creates batched sql statements. For unknown reasons Spring JPA nor
+     * JDBC batching works in appengine running in the cloud. It's old school but it solves our batching
+     * issue and can handle inserts up to 10,000 participants.
+     */
+    protected void saveParticipantCohortStatuses(List<ParticipantCohortStatus> participantCohortStatuses) {
         Statement statement = null;
         Connection connection = null;
-
-        String sqlTemplate = "insert into participant_cohort_status(" +
-                "birth_date, ethnicity_concept_id, gender_concept_id, race_concept_id, " +
-                "status, cohort_review_id, participant_id) " +
-                "values ('%s', %d, %d, %d, %d, %d, %d)";
-
-        String nextInsert = ", ('%s', %d, %d, %d, %d, %d, %d)";
         int index = 0;
-        int batchSize = 50;
-        String sqlStatement = "";
+        String sqlStatement = EMPTY_STRING;
 
         try {
             connection = jdbcTemplate.getDataSource().getConnection();
             statement = connection.createStatement();
-            connection.setAutoCommit(true);
+            connection.setAutoCommit(false);
 
             for (ParticipantCohortStatus pcs : participantCohortStatuses) {
-                sqlStatement = StringUtils.isEmpty(sqlStatement) ? sqlTemplate : sqlStatement + nextInsert;
+                sqlStatement = StringUtils.isEmpty(sqlStatement) ? SQL_TEMPLATE : sqlStatement + NEXT_INSERT;
```

Comment:
I would change your code to call String.format() on just the NEXT_INSERT bit. Formatting an ever-growing string is going to be much slower.

Nit: You could just start with sqlStatement set to SQL_TEMPLATE outside the loop and below, and get rid of EMPTY_STRING.